### PR TITLE
remove direct pathanames in gravity blacklist and whitelist scripts and store them in setupVars

### PR DESCRIPTION
**Pull request by tuplink**
_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

**By submitting this pull request, I confirm the following (please check boxes, eg [X]):**
- [x ] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [ x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x ] I have considered, and confirmed that this submission will be valuable to others.
- [x ] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x ] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**
- [ ] 1 (very unfamiliar)
- [ ] 2
- [ ] 3
- [ ] 4
- [x ] 5
- [ ] 6
- [ ] 7
- [ ] 8
- [ ] 9
- [ ] 10 (very familiar)

---

_{replace this line with your pull request content}_

### CODE REVIEW

**Comment by tuplink** [comment]
i have not tested this with and IPv6 setup.

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ setupVars=/etc/pihole/setupVars.conf
 
 webInterfaceGitUrl="https://github.com/pi-hole/AdminLTE.git"
 webInterfaceDir="/var/www/html/admin"
-piholeGitUrl="https://github.com/pi-hole/pi-hole.git"
+piholeGitUrl="https://github.com/tuplink/pi-hole.git"
```
Comment (dschaper) [inline automated install/basic-install.sh]: Please check the repo line.

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -58,7 +58,7 @@ else
   echo "::: Detecting the presence of the sudo utility for continuation of this install..."
 	if [ -x "$(command -v sudo)" ];then
 		echo "::: Utility sudo located."
-	  exec curl -sSL https://install.pi-hole.net | sudo bash "$@"
+	  exec sudo bash "$0" "$@"
```
Comment (dschaper) [inline automated install/basic-install.sh]: This will fail on a `curl` refire.

File: automated install/basic-install.sh (hunk 1 above)
Comment (tuplink) [inline automated install/basic-install.sh]: i fixed these how do i update?

Comment (dschaper) [inline automated install/basic-install.sh]: You already did, those were outdated commits, the latest version you have @ [2d7e316] takes care of that part.

**Comment by diginc** [comment]
You should quote all variable assignments and `[ -n ]` or `[ -z  ]` conditionals.  Some cases you can get away without quotes but the cases that do require it makes it so you should just always quote bash variables as a standard convention.

See my previous comment showing how an empty variable can trigger -n if you don't quote it: https://github.com/pi-hole/pi-hole/pull/774#discussion_r82660085

**Comment by tuplink** [comment]
@dschaper are you saying gravity.list shoould be in format        IPv4 IPv6 hostname?   it should return ipv6 host   if IPv6_address is set in setupVars.conf  if thats not the case it uses IPv4_address       but this should only apply to whitelist and blacklist

**Comment by tuplink** [comment]
going to test as we speak

**Comment by tuplink** [comment]
i need some direction.   should there be an entry for both ipv4 and ipv6? also the stable has a bug if you only select ipv6 durring installl you end up with 
 zz.zeroredirect1.com 
fe80::a00:27ff:fe54:df4e zz.zeroredirect1.com

but seeing that there is 2 entries im asuming we want 2 in the finised list file

**Comment by PromoFaux** [comment]
If both IPv4 and IPv6 are selected on install, then yes there should be an entry for each. Else there should only be an entry for whichever one was selected. Nice catch on the bug... I guess not many people run on IPv6 only! If nothing else that should be easy to patch!

**Comment by tuplink** [comment]
if we git this alll worked out i have it fixed going to test again

**Comment by PromoFaux** [comment]
@tuplink [this is my proposed fix for current bug in stable](https://github.com/pi-hole/pi-hole/blob/60d17b97f521595562cc1dba6cd13104e4f9eea1/gravity.sh#L250-L272)

**Comment by tuplink** [comment]
so i got it all working i do not have any VAR check but i can add that in way fewer lines

**Comment by tuplink** [comment]
so as of right now all apears to be workig with all records reporting ipv4 / ipv6 or both and a check if none are set

**Comment by PromoFaux** [comment]
The reason I took the approach I did, was so that the list remained in the order of :
```
ipv4 domain
ipv6 domain
ipv4 domain2
ipv6 domain2
```

rather than

```
ipv4 domain
ipv4 domain2
ipv6 domain
ipv6 domain2
```

**Comment by tuplink** [comment]
i agree but you are probably never going to nano a file with 10000 lines you are going to use grep to check what is in there. and as the file will be recreated every time gravity runs you wouldnt change anything in it. i wonder if there is any performance gains to be had with dnsmasq eather way.  to each his own. but the ipv4 and ipv6 check should happen before any files are downloaded imo.

**Comment by dschaper** [comment]
Closing as outdated.

**Pull request closed without merging**